src/libcommon/utility/utility.h

```diff
@@ -111,4 +113,38 @@ struct COMMON_EXPORT CommonUtility {
         static void extractIntFromStrVersion(const std::string &version, std::vector<int> &tabVersion);
 };
 
+struct ArgsReader {
+        template <class... Args>
+        explicit ArgsReader(Args... args) : stream(&params, QIODevice::WriteOnly) {
+            read(args...);
+        }
+        template <class T>
+        void read(const T p) {
+            stream << p;
+        }
+        template <class T, class... Args>
+        void read(const T p, Args... args) {
+            stream << p;
+            read(args...);
+        }
+        explicit operator QByteArray() const { return params; }
+        QByteArray params;
+        QDataStream stream;
+};
+
+struct ArgsWriter {
+        explicit ArgsWriter(const QByteArray &results) : stream{QDataStream(results)} {};
+        template <class T>
+        void write(T &r) {
+            stream >> r;
+        }
+        template <class T, class... Args>
+        void write(T &r, Args &...args) {
```

Comment:
The fix is : `void write(T &r, Args ...args)`-> `void write(T &r, Args &...args)`.
The reference qualifier `&` was missing, causing the second argument and the next not to be written.